lib/models/solid_lint_rule.dart

```diff
@@ -0,0 +1,16 @@
+import 'package:custom_lint_builder/custom_lint_builder.dart';
+import 'package:solid_lints/models/metric_rule.dart';
+
+/// A base class for emitting information about
+/// issues with user's `.dart` files.
+abstract class SolidLintRule<T extends Object?> extends DartLintRule {
+  /// Constructor for [SolidLintRule] model.
+  SolidLintRule(this.config) : super(code: config.lintCode);
+
+  /// Configuration for a particular rule with all the
+  /// defined custom parameters.
+  final MetricRule<T> config;
```

Comment:
Maybe we should merge SolidLintRule and MetricRule into a single entity? This would simplify the declaration of these rules also -- it will just be an override on the rule implementation.